src/components/Form/FormFields/PhoneNumberFormField.tsx

```diff
@@ -17,157 +16,146 @@ import {
   types: PhoneNumberType[];
   placeholder?: string;
   autoComplete?: string;
   disableValidation?: boolean;
 }
 
-export default function PhoneNumberFormField(props: Props) {
-  const field = useFormFieldPropsResolver(props);
-  const [error, setError] = useState<FieldError | undefined>();
-  const [country, setCountry] = useState<CountryData>({
-    flag: "🇮🇳",
-    name: "India",
-    code: "91",
-  });
-  const validator = useMemo(
-    () => PhoneNumberValidator(props.types),
-    [props.types],
-  );
-
-  const validate = useMemo(
-    () => (value: string | undefined, event: "blur" | "change") => {
-      if (!value || props.disableValidation) {
-        return;
-      }
-
-      const newError = validator(value);
+const PhoneNumberFormField = React.forwardRef<HTMLInputElement, Props>(
+  (props, ref) => {
+    const field = useFormFieldPropsResolver(props);
+    const [error, setError] = useState<FieldError | undefined>();
+    const [country, setCountry] = useState<CountryData>({
+      flag: "🇮🇳",
+      name: "India",
+      code: "91",
+    });
+    const validator = useMemo(
+      () => PhoneNumberValidator(props.types),
+      [props.types],
+    );
 
-      if (!newError) {
-        return;
-      } else if (event === "blur") {
-        return newError;
-      }
-    },
-    [props.disableValidation],
-  );
+    const validate = useMemo(
+      () => (value: string | undefined, event: "blur" | "change") => {
+        if (!value || props.disableValidation) {
+          return;
+        }
 
-  const setValue = useCallback(
-    (value: string) => {
-      value = value.replaceAll(/[^0-9+]/g, "");
-      if (value.length > 12 && value.startsWith("+910")) {
-        value = "+91" + value.slice(4);
-      }
+        const newError = validator(value);
 
-      const error = validate(value, "change");
-      field.handleChange(value);
+        if (!newError) {
+          return;
+        } else if (event === "blur") {
+          return newError;
+        }
+      },
+      [props.disableValidation],
+    );
 
-      setError(error);
-    },
-    [field, validate],
-  );
+    const setValue = useCallback(
+      (value: string) => {
+        value = value.replaceAll(/[^0-9+]/g, "");
+        if (value.length > 12 && value.startsWith("+910")) {
+          value = "+91" + value.slice(4);
+        }
 
-  const handleCountryChange = (value: CountryData): void => {
-    setCountry(value);
-    setValue(conditionPhoneCode(value.code));
-  };
+        const error = validate(value, "change");
+        field.handleChange(value);
 
-  useEffect(() => {
-    if (field.value && field.value.length > 0) {
-      if (field.value.startsWith("1800")) {
-        setCountry({ flag: "📞", name: "Support", code: "1800" });
-        return;
+        setError(error);
+      },
+      [field, validate, error],
+    );
+    useEffect(() => {
+      if (field.value!.length > 3) {
+        if (props.myfunc) {
+          props.myfunc(error);
+        }
```

Comment:
not readable / understandable. what does this do?

Reply:
https://github.com/ohcnetwork/care_fe/pull/8834#issuecomment-2439930727